src/main/java/com/neroreflex/pizza/Pizza.java

```diff
@@ -91,79 +87,91 @@ public final String getBotID() {
+        int nickLength = this.getNick().length();
         
-        // Analizza il messaggio per identificare richieste fatte al bot
-        Pattern invokeRegex = Pattern.compile("(" + this.getNick() + ")([\\s]+)([\\w]+)([\\s]*)([\\s\\S\\/\\:\\\\]*)");
-        Matcher invokeMatcher = invokeRegex.matcher(message);
-        if (invokeMatcher.matches()) {
+        // Cerco di capire se mi e' stato dato un comando
+        if ((message.charAt(0) == '!') && (message.length() > 1)) {
             // Ottieni il nome del comando
-            String command = invokeMatcher.group(3).toLowerCase();
+            int firstSpacePos = message.indexOf(' ');
+            firstSpacePos = (firstSpacePos <= -1)? message.length() : firstSpacePos;
             
-            // Questa e' la stringa con ulteriori dettagli sulla operazione da eseguire
-            String rawParamsString = invokeMatcher.group(5);
-            String[] params = rawParamsString.split("([\\s]+)");
-            Vector<String> args = new Vector<>();
+            String command = message.substring(1, firstSpacePos);
             
-            try {
-                for (int i = 0; i < params.length; i++) {
-                    if (params[i].length() > 0) args.add(params[i]);
-                }
-            } catch (Exception ex) {
-                
-            }
+            // Questa e' la stringa con ulteriori dettagli sulla operazione da eseguire
+            String rawParamsString = message.substring(firstSpacePos);
             
-            // Controlla se il bot e' stato salutato
-            if ((command.compareTo("hello") == 0) && (args.isEmpty())) {
-                this.enqueueMessage(new Message(channel, "Hello " + sender + " :)"));
-            } // Qualcuno ha chiesto aiuto?
-            else if ((command.compareTo("help") == 0) && (args.isEmpty())) {
-                this.enqueueMessage(new Message(channel, sender + " you should type: " + this.getNick() + " plugin list"));
+            // Controlla se il trancio e' presente e registrato
+            if (this.tranci.containsKey(command)) {
+                // E se lo e' piazza la richiesta che dovra' prendere in carico
+                if (this.tranci.get(command).isLoaded())
+                    this.tranci.get(command).enqueueRequest(new Request(channel, sender, rawParamsString));
+                else
+                    this.sendMessage(channel, sender + " sorry but the requested plugin is not loaded (yet)");
+            } else if (command.compareTo("help") == 0) {// E se non lo e' invia l'errore
+                this.Help(new Request(channel, sender, rawParamsString));
+            } else if (command.compareTo("info") == 0) {// E se non lo e' invia l'errore
+                this.Info(new Request(channel, sender, rawParamsString));
+            } else {
+                this.enqueueMessage(new Message(channel, "I don't know what I have to do at '" + command + "' request :("));
```

Comment:
La grammatica corretta sarebbe "to do with", e probabilmente quel "request" alla fine va tolto.